### DISCUSSION TITLE
NVPMODEL crashes when enabling CPU cores

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1381] [Q&A]
Hello,

I am having problems with the NVPMODEL command. It seems that disabling CPUs always work (ex. mode 5 to mode 6), but enabling CPUs triggers a software crash and reboot. This is what I see below. I am using the latest Dunfell git commit on all meta-tegra, meta-tegra-community, poky, meta-openembedded. However, I have my own layer that might break something here, not sure. I tried to build an off the shelf yocto image, the core-image-sato-dev build, following https://developer.ridgerun.com/wiki/index.php/Yocto_Support_for_NVIDIA_Jetson_Platforms_-_Setting_up_Yocto , which has the same commit point for meta-tegra etc as my build, and I don't see any problem with the NVPMODEL command there. I have been trying to skim my custom layer but so far without success. Something else to be noted is that the same custom layer, associated with the Gatesgarth build (Jetpack 4.4.1), works perfectly fine with NVPMODEL.

I was wondering from the message below if someone would have a clue of what can go wrong. Thanks so much!

root@NX:~# nvpmodel -q
NV Fan Mode:quiet
NV Power Mode: MODE_10W_DESKTOP
5
root@NX:~# nvpmodel -m 1
root@NX:~# nvpmodel -m 4
root@NX:~# nvpmodel -m 5
root@NX:~# nvpmodel -m 7
root@NX:~# nvpmodel -m 0
root@NX:~# nvpmodel -m 3
root@NX:~# nvpmodel -m 6
root@NX:~# nvpmodel -m 5
[   81.060020] Internal error: Oops: 9600004f [#1] PREEMPT SMP
[   81.063467] Modules linked in: cfg80211 nvgpu
[   81.065527] CPU: 2 PID: 0 Comm: swapper/2 Not tainted 4.9.337-l4t-r32.7.4+ga1cf17bc0aa0 #1
[   81.066298] Hardware name: NVIDIA Jetson Xavier NX Developer Kit (DT)
[   81.067252] task: 00000000c2d20c6d task.stack: 00000000ee8e8b67
[   81.067712] PC is at __memcpy+0x7c/0x180
[   81.068153] LR is at __copy_hyp_vect_bpi+0x5c/0x88
[   81.068670] pc : [<ffffff800848083c>] lr : [<ffffff800809185c>] pstate: 800001c9
[   81.069593] sp : ffffffc1f6b97ec0
[   81.069946] x29: ffffffc1f6b97ec0 x28: 0000000000000000 
[   81.070639] x27: ffffff800a41cfc4 x26: ffffff80080a3110 
[   81.071295] x25: ffffff80080a3100 x24: ffffff8009da3008 
[   81.072205] x23: ffffff80080a2000 x22: ffffff80080a2800 
[   81.072837] x21: ffffff80080a3100 x20: 0000000000000010 
[   81.073452] x19: ffffff80080a2080 x18: 0000000000000000 
[   81.074081] x17: 0000000000000000 x16: 0000000000000000 
[   81.074785] x15: 0000000000000000 x14: 0000000000000000 
[   81.075412] x13: 0000000000000000 x12: 0000000000000000 
[   81.076124] x11: 0000000000000000 x10: 00000032b5593519 
[   81.078447] x9 : 0000000000101122 x8 : d503201fd503201f 
[   81.084218] x7 : d503201fd503201f x6 : ffffff80080a2000 
[   81.089715] x5 : 000000000000003f x4 : 0000000000000000 
[   81.095051] x3 : 0000000000000010 x2 : 0000000000000010 
[   81.100406] x1 : ffffff80080a3110 x0 : ffffff80080a2000 
[   81.105717] 
[   81.107056] Process swapper/2 (pid: 0, stack limit = 0x00000000ee8e8b67)
[   81.113427] Call trace:
[   81.115855] [<000000009aef5445>] __memcpy+0x7c/0x180
[   81.120660] [<0000000047370233>] install_bp_hardening_cb.part.0+0x10c/0x190
[   81.127029] [<00000000f4b6ef32>] enable_tegra_smccc_arch_workaround_1+0x70/0x90
[   81.133766] [<00000000d38aac1d>] check_local_cpu_capabilities+0x78/0x1e0
[   81.140473] [<00000000e4e22e3d>] secondary_start_kernel+0xbc/0x160
[   81.146338] [<0000000034c40cc6>] 0x80f821a8
[   81.150144] ---[ end trace 53854ed5593b5ac2 ]---
[   81.277903] Kernel panic - not syncing: Attempted to kill the idle task!
[   81.278629] SMP: stopping secondary CPUs
[   81.279000] Kernel Offset: disabled
[   81.279225] Memory Limit: none
[   81.279444] trusty-log panic notifier - trusty version Built: 21:08:59 Jun  8 2023 [   81.377458] Rebooting in 5 seconds..
����Shutdown state requested 1
Rebooting system ...

**Comment by madisongh** [comment]
I just tried playing around with switching `nvpmodel` settings that turn off CPUs on a build off dunfell latest from our [demo distro](https://github.com/OE4T/tegra-demo-distro/tree/dunfell), and didn't see a panic.  Given that the traceback is buried down in architecture-specific code in the kernel with mixed C and assembly, it might be something related to the toolchain - gcc or binutils.  In our demo distro, we use gcc 8 on dunfell (recipes are in the `contrib` sublayer of `meta-tegra`), rather than the default version of gcc out of OE-Core.
